engine/swarm/secrets.md

```diff
@@ -0,0 +1,857 @@
+    Server public key is 4096 bit
+    Secure Renegotiation IS supported
+    Compression: NONE
+    Expansion: NONE
+    SSL-Session:
+        Protocol  : TLSv1
+        Cipher    : AES256-SHA
+        Session-ID: A1A8BF35549C5715648A12FD7B7E3D861539316B03440187D9DA6C2E48822853
+        Session-ID-ctx:
+        Master-Key: F39D1B12274BA16D3A906F390A61438221E381952E9E1E05D3DD784F0135FB81353DA38C6D5C021CB926E844DFC49FC4
+        Key-Arg   : None
+        Start Time: 1481685096
+        Timeout   : 300 (sec)
+        Verify return code: 0 (ok)
+    ```
+
+7.  To clean up after running this example, remove the `nginx` service and the
+    stored secrets.
+
+    ```bash
+    $ docker service rm nginx
+
+    $ docker secret rm site.crt site.key nginx.conf
+    ```
+
+### Advanced example: Use secrets with a WordPress service
+
+In this example, you create a single-node MySQL service with a custom root
+password, add the credentials as secrets, and create a single-node WordPress
+service which uses these credentials to connect to MySQL. The
+[next example](#example-rotate-a-secret) builds on this one and shows you how to
+rotate the MySQL password and update the services so that the WordPress service
+can still connect to MySQL.
+
+This example shows some ways to use Docker secrets to avoid saving sensitive
+credentials within your image or passing them directly on the command line.
+
+> **Note**: This example uses a single-Engine swarm for simplicity, and uses a
+> single-node MySQL service because a single MySQL server instance cannot be
+> scaled by simply using a replicated service, and setting up a MySQL cluster is
+> beyond the scope of this example.
+>
+> Also, changing a MySQL root passphrase isn’t as simple as changing
+> a file on disk. You must use a query or a `mysqladmin` command to change the
+> password in MySQL.
+
+1.  Generate a random alphanumeric password for MySQL and store it as a Docker
+    secret with the name `mysql_password` using the `docker secret create`
+    command. To make the password shorter or longer, adjust the last argument of
+    the `openssl` command. This is just one way to create a relatively random
+    password. You can use another command to generate the password if you
+    choose.
+
+    > **Note**: After you create a secret, you cannot update it. You can only
+    > remove and re-create it, and you cannot remove a secret that a service is
+    > using. However, you can grant or revoke a running service's access to
+    > secrets using `docker service update`. If you need the ability to update a
+    > secret, consider adding a version component to the secret name, so that you
+    > can later add a new version, update the service to use it, then remove the
+    > old version.
+
+    ```bash
+    $ openssl rand -base64 20 | docker secret create mysql_password
+
+    l1vinzevzhj4goakjap5ya409
+    ```
+
+    The value returned is not the password, but the ID (digest) of the secret.
```

Comment:
small nit; remove `(digest)` here; the ID of secrets have no relation to the content of the secrets (otherwise they may potentially "leak" the content).